### ISSUE TITLE
Configured static clusters appear to default to V2

### ISSUE THREAD

**Issue by phlax** [envoyproxy/envoy#13817]
### description

configuring the static_resources for xDS to consume dynamic config, i `config_dump` ed the `static_clusters` and the cluster is shown as `V2` 

not sure if there is something i could add to tell it to use v3, but either way the default seems to be v2

### steps to reproduce

start envoy with the following config

```yaml
node:
  cluster: test-cluster
  id: test-id

dynamic_resources:
  ads_config:
    api_type: GRPC
    transport_api_version: V3
    grpc_services:
    - envoy_grpc:
        cluster_name: xds_cluster
  cds_config:
    resource_api_version: V3
    ads: {}
  lds_config:
    resource_api_version: V3
    ads: {}

static_resources:
  clusters:
  - connect_timeout: 1s
    type: strict_dns
    http2_protocol_options: {}
    name: xds_cluster
    load_assignment:
      cluster_name: xds_cluster
      endpoints:
      - lb_endpoints:
        - endpoint:
            address:
              socket_address:
                address: go-control-plane
                port_value: 18000

admin:
  access_log_path: /tmp/admin_access.log
  address:
    socket_address:
      protocol: TCP
      address: 0.0.0.0
      port_value: 19000

```

dump the config as follows

```console
$  curl -s http://localhost:19000/config_dump     | jq -r '.configs[1].static_clusters'
```
you should get the following output

```json
[
  {
    "cluster": {
      "@type": "type.googleapis.com/envoy.api.v2.Cluster",
      "name": "xds_cluster",
      "type": "STRICT_DNS",
      "connect_timeout": "1s",
      "http2_protocol_options": {},
      "load_assignment": {
        "cluster_name": "xds_cluster",
        "endpoints": [
          {
            "lb_endpoints": [
              {
                "endpoint": {
                  "address": {
                    "socket_address": {
                      "address": "go-control-plane",
                      "port_value": 18000
                    }
                  }
                }
              }
            ]
          }
        ]
      }
    },
    "last_updated": "2020-10-29T10:24:52.948Z"
  }
]
```

**Comment by dio** [comment]
I think it is related to https://github.com/envoyproxy/envoy/commit/233838a39fb8310ea145e7d758d428d5a37b0306 which introduced `API_RECOVER_ORIGINAL` (and used here: https://github.com/envoyproxy/envoy/blob/233838a39fb8310ea145e7d758d428d5a37b0306/source/common/upstream/cluster_manager_impl.cc#L879). The behavior of it can be observed here: https://github.com/envoyproxy/envoy/blob/233838a39fb8310ea145e7d758d428d5a37b0306/test/common/config/version_converter_test.cc#L51-L54.

**Comment by mk46** [comment]
@phlax I think `--bootstrap-version 3`  will help you to config it as v3.

**Comment by phlax** [comment]
thanks @mk46 thats really helpful

reading the docs for the flag tho, im still thinking it should arrive at v3, unless something in the way i have configured it is preventing "upgrade":

> If unset, Envoy will attempt to load the bootstrap as the previous API version and upgrade it to the latest. If that fails, Envoy will attempt to load the configuration as the latest version.

**Comment by phlax** [comment]
i updated the config in the description as it was missing an `admin` section

i had wondered if this would be resolved by #13950 but it doesnt appear  to be
